offload/plugins-nextgen/common/include/PluginInterface.h

```diff
@@ -939,6 +939,14 @@ struct GenericDeviceTy : public DeviceAllocatorTy {
   Error printInfo();
   virtual Expected<InfoTreeNode> obtainInfoImpl() = 0;
 
+  /// Return true if the device has work that is either queued or currently
+  /// running
+  ///
+  /// Devices which cannot report this information should always return true
```

Comment:
We could use this to implement the equivalent of the `UR_QUEUE_INFO_EMPTY` query. If that's the case though, returning `true` when the device doesn't support the query would be misleading. It could return an error that gets safely handled in `olCreateEvent_impl` instead.

Reply:
Yep, I was thinking of adding an `EMPTY` queue info, but figured it would be better as a seperate change.

Returning true if it's unsupported isn't great, but I assume there are backends we wish to support that don't support these kinds of queries. I'm not sure what better way there is.

Reply:
Probably not worth the added complexity.